src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -667,7 +667,8 @@ private void WriteNonceCookie(string nonce)
                 new CookieOptions
                 {
                     HttpOnly = true,
-                    Secure = Request.IsHttps
+                    Secure = Request.IsHttps,
+                    Expires = DateTime.UtcNow + Options.ProtocolValidator.NonceLifetime
```

Comment:
Curious (not directly related to this PR): is there a reason why `CookieOptions.Expires` is not a `DateTimeOffset`?

I guess it would be also ideal to use `ISystemClock` here.